apps/design/backend/src/store.ts

```diff
@@ -1931,4 +1938,99 @@ export class Store {
       `)
     );
   }
+
+  async getQuickResultsReportingTalliesForElection(
+    election: ElectionRecord,
+    isLive: boolean
+  ): Promise<QuickResultsReportingTally[]> {
+    assert(
+      election.lastExportedBallotHash !== undefined,
+      'Election has not yet been exported.'
+    );
+    const row = (
+      await this.db.withClient((client) =>
+        client.query(
+          `
+            select
+              election_id as "electionId",
+              encoded_compressed_tally as "encodedCompressedTally",
+              is_live_mode as "isLive",
+              signed_at as "signedAt",
+              precinct_id as "precinctId",
+              machine_id as "machineId"
+            from results_reports
+            where
+              ballot_hash = $1 and
+              election_id = $2 and
+              is_live_mode = $3
+            order by signed_at desc
+          `,
+          election.lastExportedBallotHash,
+          election.election.id,
+          isLive
+        )
+      )
+    ).rows as QuickResultsReportingTallyRow[];
+    return row.map(
+      (r): QuickResultsReportingTally => ({
+        electionId: r.electionId,
+        encodedCompressedTally: r.encodedCompressedTally,
+        machineId: r.machineId,
+        isLive: r.isLive,
+        signedTimestamp: new Date(r.signedAt),
+        precinctId: (r.precinctId as unknown as PrecinctOrSplitId) || undefined,
+      })
+    );
+  }
+
+  // Save the provided quick results report, overwriting one for the given election ballot hash, machine and isLive toggle if one already exists.
+  async saveQuickResultsReportingTally({
+    electionId,
+    ballotHash,
+    encodedCompressedTally,
+    machineId,
+    isLive,
+    signedTimestamp,
+    precinctId,
+  }: {
+    electionId: string;
+    ballotHash: string;
+    encodedCompressedTally: string;
+    machineId: string;
+    isLive: boolean;
+    signedTimestamp: Date;
+    precinctId?: string;
+  }): Promise<void> {
+    await this.db.withClient((client) =>
+      client.withTransaction(async () => {
+        const { rowCount } = await client.query(
+          `
+            insert into results_reports (
+              ballot_hash,
+              election_id,
+              machine_id,
+              is_live_mode,
+              signed_at,
+              encoded_compressed_tally,
+              precinct_id
+            ) values ($1, $2, $3, $4, $5, $6, $7)
+            on conflict (ballot_hash, machine_id, is_live_mode)
+            do update set
```

Comment:
Noting that this does not enforce that the record being updated has the same `election_id` as the record we intended to create.